cupy/statistics/order.py

```diff
@@ -113,4 +114,114 @@ def nanmax(a, axis=None, out=None, keepdims=False):
 # TODO(okuta): Implement ptp
 
 
-# TODO(okuta): Implement percentile
+def percentile(a, q, axis=None, out=None, interpolation='linear',
+               keepdims=False):
+    """Compute the qth percentile of the data along the specified axis.
+
+    Args:
+        a (cupy.ndarray): Array for which to compute percentiles.
+        q (float, tuple of floats or cupy.ndarray): Percentiles to compute
+            in the range between 0 and 100 inclusive.
+        axis (int or tuple of ints): Along which axis or axes to compute the
+            percentiles. The flattened array is used by default.
+        out (cupy.ndarray): Output array.
+        interpolation (str): Interpolation method when a quantile lies between
+            two data points. ``linear`` interpolation is used by default.
+            Supported interpolations are``lower``, ``higher``, ``midpoint``,
+            ``nearest`` and ``linear``.
+        keepdims (bool): If ``True``, the axis is remained as an axis of
+            size one.
+
+    Returns:
+        cupy.ndarray: The percentiles of ``a``, along the axis if specified.
+
+    .. seealso:: :func:`numpy.percentile`
+
+    """
+    q = cupy.asarray(q, dtype=a.dtype)
+    if q.ndim == 0:
+        q = q[None]
+        zerod = True
+    else:
+        zerod = False
+    if q.ndim > 1:
+        msg = 'Expected q to have a dimension of 1.\n' \
+            'Actual: {0} != 1'.format(q.ndim)
+        raise ValueError(msg)
+
+    if keepdims:
+        if axis is None:
+            keepdim = (1,) * a.ndim
+        else:
+            keepdim = list(a.shape)
+            for ax in axis:
+                keepdim[ax % a.ndim] = 1
+            keepdim = tuple(keepdim)
+
+    # Copy a since we need it sorted but without modifying the original array
+    if isinstance(axis, int):
+        axis = axis,
+    if axis is None:
+        ap = a.flatten()
+        nkeep = 0
+    else:
+        # Reduce axes from a and put them last
+        axis = tuple(ax % a.ndim for ax in axis)
+        keep = set(range(a.ndim)) - set(axis)
+        nkeep = len(keep)
+        for i, s in enumerate(sorted(keep)):
+            a = a.swapaxes(i, s)
+        ap = a.reshape(a.shape[:nkeep] + (-1,)).copy()
+
+    axis = -1
+    ap.sort(axis=axis)
+    Nx = ap.shape[axis]
+    indices = q * 0.01 * (Nx - 1.)  # percents to decimals
+
+    if interpolation == 'lower':
+        indices = cupy.floor(indices).astype(cupy.int32)
+    elif interpolation == 'higher':
+        indices = cupy.ceil(indices).astype(cupy.int32)
+    elif interpolation == 'midpoint':
+        indices = 0.5 * (cupy.floor(indices) + cupy.ceil(indices))
+    elif interpolation == 'nearest':
+        # TODO(hvy): Implement nearest using around
+        raise NotImplementedError()
```

Comment:
`NotImplementedError` is used when an abstract class does not have an implementation. Use `ValueError`.
https://docs.python.org/3.6/library/exceptions.html#NotImplementedError

Reply:
Fixed.